src/main/java/org/apache/cassandra/sidecar/utils/SidecarVersionProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.utils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.cassandra.sidecar.common.NodeSettings;
+
+/**
+ * Retrieves, caches, and provides build version of this Sidecar binary
+ */
+public class SidecarVersionProvider
+{
+    private final String sidecarVersion;
+
+    public SidecarVersionProvider(String resource)
+    {
+        try (InputStream input = NodeSettings.class.getResourceAsStream(resource);
```

Comment:
Import  `NodeSettings` does not seem to be necessary.

Reply:
Thanks, replaced with a call to `getClass()`.